ui/src/utils/jupyterhub.ts

```diff
@@ -131,13 +139,21 @@ export const getPinnedApps = (servers: any, username: string) => {
       name: 'VSCode',
       description: 'This is your default VSCode server.',
       framework: 'VSCode',
-      url: `/hub/user/${username}/vscode`,
+      url: getEncodedServerUrl(username, 'vscode'),
       thumbnail: VSCODE_LOGO,
     });
   }
   return pinnedApps;
 };
 
+/**
+ * Get a server URL for the given username and server type, with required jupyterhub encoding.
+ */
+export const getEncodedServerUrl = (username: string, serverType: string) => {
+  const encodedUsername = username.replace(/\+/g, '%2B'); // Encode '+' with url encoding
+  return `/hub/user/${encodedUsername}/${serverType.toLowerCase()}`;
```

Comment:
Is there no native function for this?

Reply:
You can use `encodeURIComponent`, however that will also encode the `@` and wasn't sure if that made a difference. Should we go with that?

Reply:
You can encode a url (existing Nebari url) and try going to it to see if that works, like say:
https://nebari.quansight.dev/hub/user/akumar@quansight.com/vscode